Provider v3.12.0 crashes with SegFault/nil pointer dereference

### Community Note

- Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
- Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
- If you are interested in working on this issue or have submitted a pull request, please leave a comment

<!--- Thank you for keeping this note for the community --->

### Terraform Version
```
$ terraform version
Terraform v1.0.0
```

### Affected Resource(s)

<!--- Please list the affected resources and data sources. --->

- resource_okta_policy_rule_signon

### Terraform Configuration Files
Part of a larger module

```hcl
resource "okta_policy_signon" "this" {
  name            = var.customer_name
  status          = "ACTIVE"
  description     = var.customer_name
  groups_included = [okta_group.customer_group.id]
  priority        = 1

  lifecycle {
    ignore_changes = [priority]
  }
}

resource "okta_policy_rule_signon" "this" {
  policyid = okta_policy_signon.this.id
  name     = var.customer_name
  priority = 1

  lifecycle {
    ignore_changes = [priority]
  }
}
```

### Debug Output

https://gist.github.com/sklarsa/8284e188c63dcf98db23e6265f8cc1f0

### Panic Output

https://gist.github.com/sklarsa/7ade7c59f247fc6de47f6ae1b2174345

### Expected Behavior

`terraform plan` completes with exit code 0

### Actual Behavior

`terraform plan` fails

### Steps to Reproduce

<!--- Please list the steps required to reproduce the issue. --->

1. `terraform plan`

### Important Factoids

Running inside a GitLab pipeline with docker executor on an ubuntu-based image

### References

<!---
Information about referencing Github Issues: https://help.github.com/articles/basic-writing-and-formatting-syntax/#referencing-issues-and-pull-requests

Are there any other GitHub issues (open or closed) or pull requests that should be linked here? Vendor documentation? For example:
--->

After some very preliminary investigation (and pardon my unfamiliarity with the library), could this be the issue?

Stack Trace:
```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0xf2fb1d]
goroutine 135 [running]:
github.com/okta/terraform-provider-okta/okta.resourcePolicySignOnRuleRead(0x13a5eb8, 0xc00096a960, 0xc000421600, 0x103c480, 0xc00019a540, 0xc000214470, 0xc0001c3908, 0x40e0f8)
	github.com/okta/terraform-provider-okta/okta/resource_okta_policy_rule_sign_on.go:163 +0x43d
...
```

https://github.com/okta/terraform-provider-okta/blob/63b82d9ba7648574ec9866b3a66ff48f3ac65c4e/okta/resource_okta_policy_rule_sign_on.go#L163

`rule` is instantiated by `getPolicyRule`: https://github.com/okta/terraform-provider-okta/blob/63b82d9ba7648574ec9866b3a66ff48f3ac65c4e/okta/policy_rule.go#L165

`getPolicyRule` returns a `PolicyRule` object, that has a `Conditions` attribute of type `*okta.PolicyRuleConditions`:  https://github.com/okta/terraform-provider-okta/blob/63b82d9ba7648574ec9866b3a66ff48f3ac65c4e/sdk/policy_rule.go#L24

https://github.com/okta/okta-sdk-golang/blob/a5b26e82ebd9fbeb45e04bcd002f2629fcb47660/okta/policyRuleConditions.go#L38

If the server never returns a `RiskLevel` (which I don't see in responses from the rules endpoint on my admin dashboard), it's possible that `Conditions.RiskLevel == nil`, although I'm not quite sure what the `omitempty` annotation does

FWIW I can reproduce this behavior on my local mac as well

--- a/okta/resource_okta_app_oauth.go
+++ b/okta/resource_okta_app_oauth.go
@@ -197,7 +197,7 @@ func resourceAppOAuth() *schema.Resource {
 			"wildcard_redirect": {
 				Type:             schema.TypeString,
 				Optional:         true,
-				Description:      "Indicates if the client is allowed to use wildcard matching of redirect_uris",
+				Description:      "*Early Access Property*. Indicates if the client is allowed to use wildcard matching of redirect_uris",
 				Default:          "DISABLED",
 				ValidateDiagFunc: elemInSlice([]string{"DISABLED", "SUBDOMAIN"}),
 			},
@@ -464,7 +464,9 @@ func resourceAppOAuthRead(ctx context.Context, d *schema.ResourceData, m interfa
 	_ = d.Set("tos_uri", app.Settings.OauthClient.TosUri)
 	_ = d.Set("policy_uri", app.Settings.OauthClient.PolicyUri)
 	_ = d.Set("login_uri", app.Settings.OauthClient.InitiateLoginUri)
-	_ = d.Set("wildcard_redirect", app.Settings.OauthClient.WildcardRedirect)
+	if app.Settings.OauthClient.WildcardRedirect != "" {
+		_ = d.Set("wildcard_redirect", app.Settings.OauthClient.WildcardRedirect)
+	}
 	_ = d.Set("auto_submit_toolbar", app.Visibility.AutoSubmitToolbar)
 	_ = d.Set("hide_ios", app.Visibility.Hide.IOS)
 	_ = d.Set("hide_web", app.Visibility.Hide.Web)

--- a/okta/resource_okta_group.go
+++ b/okta/resource_okta_group.go
@@ -105,6 +105,10 @@ func resourceGroupDelete(ctx context.Context, d *schema.ResourceData, m interfac
 }
 
 func syncGroupUsers(ctx context.Context, d *schema.ResourceData, m interface{}) error {
+	// Only sync when the user opts in by outlining users in the group config
+	if _, exists := d.GetOk("users"); !exists {
+		return nil
+	}
 	userIDList, err := listGroupUserIDs(ctx, m, d.Id())
 	if err != nil {
 		return err

--- a/okta/resource_okta_policy_rule_sign_on.go
+++ b/okta/resource_okta_policy_rule_sign_on.go
@@ -160,7 +160,9 @@ func resourcePolicySignOnRuleRead(ctx context.Context, d *schema.ResourceData, m
 	if rule.Actions.SignOn.FactorPromptMode != "" {
 		_ = d.Set("mfa_prompt", rule.Actions.SignOn.FactorPromptMode)
 	}
-	_ = d.Set("risc_level", rule.Conditions.RiskScore.Level)
+	if rule.Conditions != nil && rule.Conditions.RiskScore != nil {
+		_ = d.Set("risc_level", rule.Conditions.RiskScore.Level)
+	}
 	err = setNonPrimitives(d, map[string]interface{}{
 		"behaviors": convertStringSetToInterface(rule.Conditions.Risk.Behaviors),
 	})

--- a/okta/user_schema.go
+++ b/okta/user_schema.go
@@ -199,8 +199,6 @@ func syncBaseUserSchema(d *schema.ResourceData, subschema *sdk.UserSubSchema) {
 	}
 	if len(subschema.Permissions) > 0 {
 		_ = d.Set("permissions", subschema.Permissions[0].Action)
-	} else {
-		_ = d.Set("permissions", "")
 	}
 	if subschema.Pattern != nil {
 		_ = d.Set("pattern", &subschema.Pattern)

--- a/website/docs/r/app_oauth.html.markdown
+++ b/website/docs/r/app_oauth.html.markdown
@@ -75,7 +75,7 @@ The following arguments are supported:
 
 - `redirect_uris` - (Optional) List of URIs for use in the redirect-based flow. This is required for all application types except service.
 
-- `wildcard_redirect` - (Optional) Indicates if the client is allowed to use wildcard matching of `redirect_uris`. Valid values: `"DISABLED"`, `"SUBDOMAIN"`. Default value is `"DISABLED"`.
+- `wildcard_redirect` - (Optional) *Early Access Property*. Indicates if the client is allowed to use wildcard matching of `redirect_uris`. Valid values: `"DISABLED"`, `"SUBDOMAIN"`. Default value is `"DISABLED"`.
 
 - `post_logout_redirect_uris` - (Optional) List of URIs for redirection after logout.
 

